src/main/java/uk/nhs/tis/sync/service/DmsRecordAssembler.java

```diff
@@ -56,18 +67,55 @@ private DmsDto assembleDmsDto(Object dto) {
       if (mapperClass != null) {
         DmsMapper<?, ?> dmsMapper = Mappers.getMapper(mapperClass);
         dmsData = dmsMapper.objectToDmsDto(dto);
+        if (dto instanceof ProgrammeMembershipDTO) {
+          hasMultipleDmsDtos = true;
+          dmsDtos = buildCurriculumMembershipDmsDtos(schema, table, (ProgrammeMembershipDTO) dto,
+              (CurriculumMembershipDmsDto) dmsData);
+        }
       } else {
         dmsData = dto;
       }
     }
 
-    if (dmsData != null) {
-      MetadataDto metadata = new MetadataDto(Instant.now().toString(), DATA, LOAD,
-          PARTITION_KEY_TYPE, schema, table, UUID.randomUUID().toString());
+    if (hasMultipleDmsDtos) {
+      return dmsDtos;
+    } else {
+      if (dmsData != null) {
+        MetadataDto metadata = new MetadataDto(Instant.now().toString(), DATA, LOAD,
+            PARTITION_KEY_TYPE, schema, table, UUID.randomUUID().toString());
 
-      return new DmsDto(dmsData, metadata);
+        return Collections.singletonList(new DmsDto(dmsData, metadata));
+      }
```

Comment:
A rethink of the DmsMapper interface might help here.

If we changed `O objectToDmsDto(Object object)` to `List<O> objectToDmsDtos(Object object)` we could expand the DTOs inside the mapper, then create the `DmsDto` including metadata for each item in the list.

Reply:
The mapper would be more complex but it feels an appropriate place to do it.
The alternative to keep the single DTO mappers simple could be to add a `multipleDto` flag to the enum and call `objectToDmsDto` or `objectToDmsDtos` based on that.

Reply:
Let me take a look, that sounds like a good idea.

Reply:
That's pretty much it yeah, added a small comment where an update mapper could tidy up a few lines.